overlord/snapstate/snapstate.go

```diff
@@ -314,10 +311,73 @@ func Remove(s *state.State, name string) (*state.TaskSet, error) {
 	return full, nil
 }
 
-// Rollback returns a set of tasks for rolling back a snap.
+// Revert returns a set of tasks for reverting to the pervious version of the snap.
 // Note that the state must be locked by the caller.
-func Rollback(s *state.State, snap, ver string) (*state.TaskSet, error) {
-	return nil, fmt.Errorf("rollback not implemented")
+func Revert(s *state.State, name string) (*state.TaskSet, error) {
+	var snapst SnapState
+	err := Get(s, name, &snapst)
+	if err != nil && err != state.ErrNoState {
+		return nil, err
+	}
+
+	pi := snapst.PreviousSideInfo()
+	if pi == nil {
+		return nil, fmt.Errorf("no revision to revert to")
+	}
+	return revertToRevision(s, name, pi.Revision)
+}
+
+func revertToRevision(s *state.State, name string, rev snap.Revision) (*state.TaskSet, error) {
```

Comment:
I'd expect revert to have a lot in common with a standard install, instead of cooking it all again. Isn't it exactly an install, except we don't need to copy data over? Sorry, you've been on that for a while so I'm sure there are details which I'm missing, but I'd really like to make this less special and use more of the common path.